device/src/esp32_run.rs

```diff
@@ -151,6 +151,7 @@ where
                 upstream_connection.set_state(UpstreamConnectionState::PowerOn, &mut ui);
                 next_write_magic_bytes_downstream = Instant::from_ticks(0);
                 upgrade = None;
+                ui.set_device_name(name.clone());
```

Comment:
Curious as to why this was needed?

Reply:
The ui might not have the fixed name of the device when a soft reset is triggered.